desafio2.js

```diff
@@ -44,19 +48,18 @@ throw new Error(error)        }
         try{
         const data=await fs.promises.readFile(this.archivo, "utf-8")
```

Comment:
la logica adentro del try es la siguiente:
            const data=await fs.promises.readFile(this.archivo,"utf-8");
            const jsonData=JSON.parse(data)
            const deleted = jsonData.find(i=>i.id==id)
            const products = jsonData.filter(i=>i.id!=id)
            await fs.promises.writeFile(this.archivo,JSON.stringify(products,null,2))            
            return deleted